portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,145 @@
+            content = getParameter(request, "text-input", "");
+            owo = getBooleanParameter(request, "owo", false);
+            uwu = getBooleanParameter(request, "uwu", false);
+            DataServlet.numberOfShowedComments = getIntParameter(request, "num-input", 3);
+        } catch (NumberFormatException nfe) {
+            response.sendError(400, "Comment amount must be a number");
+            return;
+        } catch (Exception e){
+            response.sendError(500, "Error");
+            return;
+        }
+
+        String[] commentWords = content.split(" ", 0);
+        String commentConcatenated = "";
+        for(String word : commentWords){
+            if(owo){
+                word += "owo";
+            } else if (uwu){
+                word += "uwu";
+            } 
+            commentConcatenated += word + " ";
+        }
+
+        Entity commentEntity = new Entity("Comment");
+        commentEntity.setProperty("name", name);
+        commentEntity.setProperty("content", commentConcatenated);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
+
+        response.sendRedirect("/index.html");
+    }
+
+    /**
+     * Helper method to convert an ArrayList of Comments to JSON format
+     * @param ArrayList<Comment> comments to be converted
+     * @return JSON as a String
+     */
+    private String convertToJsonUsingGson(ArrayList<Comment> comments) {
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
+        return json;
+    }
+
+    /**
+     * @return the request parameter, or the default value if the parameter
+     *         was not specified by the client
+     */
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+        String value = request.getParameter(name);
+        if (value == null) {
+            return defaultValue;
+        }
+        return value;
+    }
+
+    /**
+     * @return the request parameter as an int, or the default value if the parameter
+     *         was not specified by the client
+     */
+    private int getIntParameter(HttpServletRequest request, String userValue, int defaultValue) {
+        String toParse = request.getParameter(userValue);
+        if ("".equals(toParse) || toParse == null) {
+            return defaultValue;
+        }
+        int value = Integer.parseInt(toParse);
+        if (value < 0) {
+            return defaultValue;
+        }
+        return value;
+    }
+
+    /**
+     * @return the request parameter as a boolean, or the default value if the parameter
+     *         was not specified by the client
+     */
+    private Boolean getBooleanParameter(HttpServletRequest request, String userValue, boolean defaultValue) {
+        Boolean value = defaultValue;
+        value = Boolean.parseBoolean(request.getParameter(userValue));
+        if (value == null) {
```

Comment:
I dont think parseBoolean ever returns null [(docs).](https://docs.oracle.com/javase/8/docs/api/java/lang/Boolean.html#parseBoolean-java.lang.String-) Is the idea to check whether the parameter is null or the parsed value?

Reply:
Yeah, parse Boolean doesn't return a null. I was aiming to get the parameter value but I must have forgot.